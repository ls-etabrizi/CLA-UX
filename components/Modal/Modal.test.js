import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Modal } from '.'

afterEach(cleanup)

describe('Modal Component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Modal open />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders some content, when passed as children', () => {
    const { getByText } = render(
      <Modal open>
        <p>Content</p>
      </Modal>
    )
    const content = getByText('Content')
    expect(content).toBeTruthy()
  })
  it('closes the modal when a function is passed to close it', () => {
    const { container } = render(
      <Modal
        open
        onClick={() => {
          document.querySelector('.modal').classList.remove('modal--open')
        }}
      />
    )
    const modal = container.querySelector('.modal')
    const modalClose = container.querySelector('.modal__close')
    fireEvent.click(modalClose)
    expect(modal).toHaveClass('modal')
  })
  it('fires an onClick function, when esc key is pressed', () => {
    const { container } = render(
      <Modal
        open
        onClick={() => {
          document.querySelector('.modal').classList.add('test')
        }}
      />
    )
    const modal = container.querySelector('.modal')
    fireEvent.keyDown(modal, {
      key: 'Escape',
      code: 'Escape',
      keyCode: 27,
      charCode: 27
    })
    expect(modal).toHaveClass('test')
  })
})
