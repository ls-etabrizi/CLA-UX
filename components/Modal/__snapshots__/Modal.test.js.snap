// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="modal-overlay modal-overlay--open"
    />
    <div
      class="modal modal--open  "
    >
      <div
        class="modal__content"
      >
        <button
          aria-label="Close modal"
          class="modal__close"
        >
          Ã—
        </button>
        <div />
      </div>
    </div>
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</DocumentFragment>
`;
